addons/gamification/models/goal.py

```diff
@@ -293,7 +293,7 @@ def update_goal(self):
                 Obj = self.env[definition.model_id.model]
 
                 field_date_name = definition.field_date_id.name
-                if definition.computation_mode == 'count' and definition.batch_mode:
+                if definition.batch_mode:
```

Comment:
more prudent to keep compute_mode in sum, count

even if not visible in web client, it could be set wrongly and apply wrongly the sum code here

Reply:
Added condition where the count/sum assumption was made, and added a logged error if an unknown computation mode is encountered